### DISCUSSION TITLE
Problema configurazione caddy (reverse proxy) utilizzando duckdns FQDN

### DISCUSSION THREAD

**Discussion by BarbaRossiccia** [indomus/forum#3246] [HUB: Home Assistant]
Salve a tutti ragazzi,  grazie per il meraviglioso lavoro che fate, sono un neofita che  si sta appassionando, sempre di piú al mondo della domotica casalinga.
Il mio scopo sarebbe quello di controllare tramite alexa qualche automazione della domotica in casa, riconosciuta da HA (cambiare canale alla tv... tramite il metodo "gratuito"), ho settato HA su un notebook inutilizzato, trasformato in server e sto utilizzando la versione core su Debian 12 tramite docker.
Sto incontrando problemi però seguendo questa guida: 
[https://indomus.it/guide/installare-e-configurare-caddy-reverse-proxy-con-docker-su-linux-debian/](url)
Allora nella mia configurazione ho un router fritzbox 7530ax che si occupa di aggiornare tramite DynDNS interno l'associazione tra il suo ip e il FQDN (duckdns).
Quando vado a configurare caddy come container in docker, come descritto minuziosamente nella guida, lo lancio e il log da molteplici errori, come se non riuscisse a scaricare i certificati.
`{"level":"error","ts":1728046671.0499086,"logger":"tls.issuance.acme.acme_client","msg":"challenge failed","identifier":"mioindirizzo.duckdns.org","challenge_type":"tls-alpn-01","problem":{"type":"urn:ietf:params:acme:error:connection","title":"","detail":"xxx.xxxx.xxxx.xxxx mio indirizzo ip: Timeout during connect (likely firewall problem)","instance":"","subproblems":[]}}
{"level":"error","ts":1728046671.049988,"logger":"tls.issuance.acme.acme_client","msg":"validating authorization","identifier":"mioindirizzo.duckdns.org","problem":{"type":"urn:ietf:params:acme:error:connection","title":"","detail":"xxx.xxxx.xxxx.xxxx mio indirizzo ip: Timeout during connect (likely firewall problem)","instance":"","subproblems":[]},"order":"https://acme-staging-v02.api.letsencrypt.org/acme/order/165837503/19549622203","attempt":1,"max_attempts":3}`

(ho postato solo i primi due errori, in quanto c'erano il nome del dominio e l'indirizzo ip e avrei dovuto cambiarli tutti a mano, se mi potete consigliare uno script, programmino c o pyton per cambiare nomedominio e ip in tutto il log, ve ne sarei grato, ma non vorrei andare ad impelagarmi in un altro problema in quanto non sono esperto neanche di programmazione ed una cosa alla volta).
Sto impazzendo! All'inizio pensavo di aver sbagliato il port forwarding nel router, ma ho aperto le porte 443 e 80 per l'ottenimento dei certificati ssl da Let’s Encrypt (per le challenge, documentatomi grazie alla guida, nel mio piccolo sto imparando tante cose nuove), ho aperto anche le porte nel firewall in debian, sulla mia macchina è iptables, tant'è che dando i comandi sul terminale:
`nmap -p 443 localhost`
`nmap -p 80 localhost`
mi dice che le porte sono aperte.
Spero sappiate aiutarmi e scusate l'ignoranza, sto cercando di impegnarmi al massimo, stavo valutando sempre da vostre guide di usare godaddy (ho visto un dominio per 1 anno a poco piú di un euro, per quello che ci devo fare top)penso si chiamasse così e cloudfare per il tunneling, però è un fatto di principio ormai vorrei farlo funzionare prima con questo metodo, sperando che non sia un problema di duckdns, ma non credo perchè sul dominio mi fa arrivare instradandomi sulla pagina del mio router, però ovviamente non essendoci certificati, firefox, mi dice che è un sito malevolo e per andare avanti bisogna premere accetto, qualcosa del genere.
Grazie ancora.

**Comment by snakuzzo** [comment]
Quel comando nmap serve a niente, anche perché è totalmente sbagliato.
Non è detto che puntando la loopback con localhost tu ottenga lo stesso risultato ottenuto puntando l'IP LAN assegnato al server. 
Poi...vedendo gli errori, c'è poco da fare...e il problema devi (purtroppo per te) cercarlo nella configurazione del port forwarding e del server locale.
Per far funzionare la http challenge e/o la tls-alpn challenge deve esserci il port forwarding correttamente funzionante della 80 per la challenge http e/o della 443 per la challenge tls-alpn.

Il giro è...

```
INTERNET -> IP PUBBLICO -> ROUTER -> PORT FORWARDING 443 (e/o 80) -> SERVER CADDY <PORTA CADDY> -> SERVIZIO <PORTA SERVIZIO>
```

Comunque...soprattutto se sei poco esperto, ma non solo per questo, ti conviene usare il tuunel di Cloudflare

**Reply by BarbaRossiccia** [reply]
Innanzitutto grazie per la risposta celere e per le spiegazioni. Certamente il problema sarà nella configurazione del server locale o del port forwarding.
Come da te suggerito mi sa che cercherò di implementare il tunnel con Cloudflare, anche perchè non esponendo porte mi sembra molto piu' sicuro. Grazie ancora.
